witness/src/Dscp/Witness/Workers/Worker.hs

```diff
@@ -18,11 +22,24 @@ import Dscp.Snowdrop.Mode
 import Dscp.Util
 import Dscp.Witness.Launcher.Mode
 import Dscp.Witness.Logic
+import Dscp.Witness.Mempool
 import Dscp.Witness.Messages
+import Dscp.Witness.Util (dieGracefully)
 
+newtype FailedTxs = FailedTxs (TVar (HashMap (Hash GTxWitnessed) GTxWitnessed))
 
-witnessWorkers :: WitnessWorkMode ctx m => [Worker m]
-witnessWorkers = [blockUpdateWorker]
+newFailedTxs :: MonadIO m => m FailedTxs
+newFailedTxs = FailedTxs <$> newTVarIO mempty
+
+witnessWorkers
+    :: WitnessWorkMode ctx m
+    => TxRelayInput
```

Comment:
These two arguments are available from the context.

Reply:
You're right!

Reply:
We're not in the `m` monad here, so to use contect we have to wrap all into it. Should we really do it?

Reply:
Also, shoud we put `failedTxs` into context? And, maybe, group it and pipes into `RelayState`?